src/core_modules/capture-core/components/Pages/Enrollment/EnrollmentPage.component.js

```diff
@@ -22,8 +44,65 @@ const getStyles = ({ typography }) => ({
     },
 });
 
+const Extras = ({ width, classes }) => {
+    const dispatch = useDispatch();
+    const { selectedProgramId, selectedEnrollmentId } = useUrlQueries();
+    const { enrollments, teiDisplayName, tetId } = useSelector(({ enrollmentPage }) => enrollmentPage);
+    const enrollmentsAsOptions = buildEnrollmentsAsOptions(enrollments, selectedProgramId);
+    const { trackedEntityName } = getScopeInfo(tetId);
+    const enrollmentLockedSelectReady = Array.isArray(enrollments);
+
+    const onTeiSelectionReset = useCallback(
+        () => {
+            dispatch(resetTeiSelection());
+        },
+        [dispatch]);
+
+    const onEnrollmentSelectionSet = useCallback(
+        (enrollmentId) => {
+            dispatch(setEnrollmentSelection({ enrollmentId }));
+        },
+        [dispatch]);
+
+    const onEnrollmentSelectionReset = useCallback(
+        () => {
+            dispatch(resetEnrollmentSelection());
+        },
+        [dispatch]);
+
+    return (<>
+        <Grid item xs={12} sm={width * 3} md={width * 2} lg={2} className={classes.orgUnitSelector}>
+            <SingleLockedSelect
+                ready={enrollmentLockedSelectReady}
+                onClear={onTeiSelectionReset}
+                options={[{
+                    label: teiDisplayName,
+                    value: 'alwaysPreselected',
+
+                }]}
+                selectedValue="alwaysPreselected"
+                title={trackedEntityName}
+            />
+        </Grid>
+        {
+            enrollmentsAsOptions &&
+            <Grid item xs={12} sm={width * 3} md={width * 2} lg={2} className={classes.orgUnitSelector}>
+                <SingleLockedSelect
+                    onClear={onEnrollmentSelectionReset}
+                    ready={enrollmentLockedSelectReady}
+                    onSelect={onEnrollmentSelectionSet}
+                    options={enrollmentsAsOptions}
+                    selectedValue={selectedEnrollmentId}
+                    title={i18n.t('Enrollment')}
+                />
+            </Grid>
+        }
+    </>);
+};
+
+
 const EnrollmentPagePlain = ({ classes, enrollmentPageStatus }) => (<>
-    <LockedSelector pageToPush="enrollment" />
+    <LockedSelector pageToPush="enrollment" renderExtraSelectors={(width, extraClasses) => <Extras width={width} classes={extraClasses} />} />
```

Comment:
I would prefer wrapping `renderExtraSelectors` in `useCallback`. I still think it is good practice to use the same reference whenever possible.